File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}

        for i in strs:
            sorted_i = sorted(i)
            sorted_i = "".join(sorted_i)
            if sorted_i in hashmap:
                hashmap[sorted_i].append(i)
            else:
                hashmap[sorted_i] = [i]
        
        res = []
        for i in hashmap:
            res.append(hashmap[i])
        return res

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hm = new HashMap<>();
        List<List<String>> answer = new ArrayList<>();

        for(String s : strs) {
            char[] chararray = s.toCharArray();
            Arrays.sort(chararray);
            String temp_str = new String(chararray);

            if(hm.containsKey(temp_str)) {
              List<String> temp_lst = hm.get(temp_str);
              temp_lst.add(s);
              hm.put(temp_str, temp_lst);
            }
            else {
              List<String> temp_lst = new ArrayList<>();
              temp_lst.add(s);
              
              hm.put(temp_str, temp_lst);
            }
            
        }
        
        return new ArrayList<>(hm.values());
    }
}